### PR TITLE
Developer: Database Export/Import file

### PR DESCRIPTION
**Pull request by bjtitus**
* Adds file handling for a `pcasts` file
* Imports preferences and database from the file & kills the app
* Option to delete the DB & Settings file by just deleting and resetting `UserDefaults` instead of incrementally unsubscribing (as "Unsubscribe from all Podcasts" does)

https://github.com/Automattic/pocket-casts-ios/assets/3250/17e1d388-6287-452c-8814-88e2cc12bd20

#### Test Bundle
* Visit Developer Menu
* Tap "Export Bundle"
* Save in Files
* Tap the "Reset Database + Settings" - app will quit
* Visit Developer Menu
* Tap "Import Bundle" - app will quit
* Select the `pcasts` file
* Open the app and ensure podcasts and settings match

#### Test import from support export
* Go to Profile > Help & Feedback > ⋯ > Export Database
* Share the file to your computer
* Unzip the file
* Append the `pcasts` extension to the folder
* Delete and re-run the app
* Drag and drop this file onto the simulator
* Hit the "Import" option when prompted
* Podcasts and settings should match

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/52">7.66</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by leandroalonso** [comment]
@bjtitus do you still plan to submit this one? I think it would be super helpful to have that. :)

**Comment by bjtitus** [comment]
@leandroalonso Yep, I just haven't had a chance to swing back around to it but I'll try to get to it this week.

**Review by leandroalonso — APPROVED** [review]
@bjtitus it all works as described! Thanks for adding that! :chef-kiss:

I have just one question though: can't we unzip the file if necessary? I found out that I mistakenly was renaming the zip file, not the folder.

This is not a big deal though and not a blocker, it can be addressed in another PR.

File: podcasts/podcasts-Info.plist [hunk 1]
```diff
@@ -181,6 +144,31 @@
 					<string>AppIcon-Dark40x40@2x~ipad</string>
 				</array>
 			</dict>
+			<key>AppIcon-Default</key>
```
Comment (leandroalonso) [inline podcasts/podcasts-Info.plist:146]: Was these changes made by Xcode?

**Comment by bjtitus** [comment]
@leandroalonso 

> can't we unzip the file if necessary? I found out that I mistakenly was renaming the zip file, not the folder.

I could allow the app to accept `pcasts` zip files but there's no built in unzipping API, as far as I can tell. We would have to import something like [ZIPFoundation](https://github.com/weichsel/ZIPFoundation) to perform the unzipping in-app.

If I did add that, I'd probably zip up the contents of the exported `pcasts` file as well since it isn't currently zipped.

File: podcasts/podcasts-Info.plist (hunk 1 above)
Comment (bjtitus) [reply podcasts/podcasts-Info.plist:146]: Yeah, it looks like a bunch of changes happened when I added the new type identifier and document type. I believe they are mostly position changes in the file.

**Comment by leandroalonso** [comment]
> I could allow the app to accept pcasts zip files but there's no built in unzipping API, as far as I can tell. We would have to import something like [ZIPFoundation](https://github.com/weichsel/ZIPFoundation) to perform the unzipping in-app.

Yikes! Let's not do that.

Nvm! :shipit:

**Pull request merged**